analyzer/src/test/kotlin/AnalyzerResultBuilderTest.kt

```diff
@@ -158,6 +159,18 @@ class AnalyzerResultBuilderTest : WordSpec() {
 
                 serializedResult2 shouldBe serializedResult
             }
+
+            "use the dependency graph representation on serialization" {
+                val mergedResults = AnalyzerResultBuilder()
+                    .addResult(analyzerResult1)
+                    .addResult(analyzerResult2)
+                    .build()
+
+                val serializedMergedResults = yamlMapper.writeValueAsString(mergedResults)
+                val deserializedMergedResults = yamlMapper.readValue<AnalyzerResult>(serializedMergedResults)
+
+                deserializedMergedResults.dependencyGraphs.keys should containExactlyInAnyOrder("type-1", "type-2")
```

Comment:
According to the test name, shouldn't you directly check on `serializedMergedResults`? Otherwise you're rather testing the *de*serialization, which might apply its own magic.

Reply:
Well, these are white-box tests; we know that the standard Jackson deserialization is used.

Testing on the serialized form is problematic. It would probably require some string matching. If you prefer, I can try this out.

Reply:
Just as an idea, another option would be to use `readTree` instead and do the assertion on the node, this would decouple the test from any deserialization logic.

Reply:
Good idea! I reworked the test case to read the JSON tree and operate on the nodes.